src/tools/cephfs/Resetter.cc

```diff
@@ -26,7 +26,38 @@
 #define dout_context g_ceph_context
 #define dout_subsys ceph_subsys_mds
 
-int Resetter::reset(mds_role_t role)
+int Resetter::init(mds_role_t role_, const std::string &type)
+{
+  role = role_;
+  int r = MDSUtility::init();
+  if (r < 0) {
+    return r;
+  }
+
+  auto fs = fsmap->get_filesystem(role.fscid);
+  assert(nullptr != fs);
+
+  is_mdlog = false;
+  if (type == "mdlog") {
+    JournalPointer jp(role.rank, fs->mds_map.get_metadata_pool());
+    int rt = jp.load(objecter);
```

Comment:
If the journal does not exist, then `--force` reset fails:

````
2018-02-07T22:40:18.467 INFO:teuthology.orchestra.run.smithi035.stderr:Error loading journal: (2) No such file or directory, pass --force to forcibly reset this journal
...
2018-02-07T22:40:18.472 INFO:tasks.cephfs_test_runner:test_rebuild_simple (tasks.cephfs.test_recovery_pool.TestRecoveryPool) ... ERROR
...
2018-02-07T22:40:45.885 INFO:tasks.cephfs_test_runner:======================================================================
2018-02-07T22:40:45.885 INFO:tasks.cephfs_test_runner:ERROR: test_rebuild_simple (tasks.cephfs.test_recovery_pool.TestRecoveryPool)
2018-02-07T22:40:45.885 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2018-02-07T22:40:45.885 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2018-02-07T22:40:45.885 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20180207.004512/qa/tasks/cephfs/test_recovery_pool.py", line 220, in test_rebuild_simple
2018-02-07T22:40:45.885 INFO:tasks.cephfs_test_runner:    self.mount_a, self.mount_b))
2018-02-07T22:40:45.885 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20180207.004512/qa/tasks/cephfs/test_recovery_pool.py", line 181, in _rebuild_metadata
2018-02-07T22:40:45.885 INFO:tasks.cephfs_test_runner:    'reset', '--force'])
2018-02-07T22:40:45.885 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20180207.004512/qa/tasks/cephfs/filesystem.py", line 1191, in journal_tool
2018-02-07T22:40:45.886 INFO:tasks.cephfs_test_runner:    return self._run_tool("cephfs-journal-tool", args, rank, quiet)
2018-02-07T22:40:45.886 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20180207.004512/qa/tasks/cephfs/filesystem.py", line 1170, in _run_tool
2018-02-07T22:40:45.886 INFO:tasks.cephfs_test_runner:    stdout=StringIO()).stdout.getvalue().strip()
2018-02-07T22:40:45.886 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/remote.py", line 193, in run
2018-02-07T22:40:45.886 INFO:tasks.cephfs_test_runner:    r = self._runner(client=self.ssh, name=self.shortname, **kwargs)
2018-02-07T22:40:45.886 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/run.py", line 423, in run
2018-02-07T22:40:45.886 INFO:tasks.cephfs_test_runner:    r.wait()
2018-02-07T22:40:45.886 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/run.py", line 155, in wait
2018-02-07T22:40:45.886 INFO:tasks.cephfs_test_runner:    self._raise_for_status()
2018-02-07T22:40:45.886 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/run.py", line 177, in _raise_for_status
2018-02-07T22:40:45.886 INFO:tasks.cephfs_test_runner:    node=self.hostname, label=self.label
2018-02-07T22:40:45.886 INFO:tasks.cephfs_test_runner:CommandFailedError: Command failed on smithi035 with status 254: u'cephfs-journal-tool --debug-mds=4 --debug-objecter=1 --rank=cephfs:0 journal reset --force'
````

From: http://qa-proxy.ceph.com/teuthology/pdonnell-2018-02-07_22:20:50-fs-wip-pdonnell-testing-20180207.004512-testing-basic-smithi/2165805/teuthology.log